documentation/rpi3.md

```diff
@@ -0,0 +1,144 @@
+  file. If you forget that, then you can get all sorts of strange errors.
+
+- From the [README.md](https://github.com/OP-TEE/optee_os#5-repo-manifests),
+  you should follow section 5.1, 5.2. In short if you have repo installed, what
+  you need to do is something like this:
+```
+$ mkdir rpi3
+$ cd rpi3
+$ repo init -u https://github.com/OP-TEE/manifest.git -m rpi3_experimental.xml
+$ repo sync -j3
+```
+  Now it's probably a good idea to read the [Tips and tricks](https://github.com/OP-TEE/optee_os#58-tips-and-tricks)
+  section also, since that will save a lot of time in the long run.
+
+- Next step is to get the toolchains
+```
+$ cd build
+$ make toolchains
+```
+
+- Then it is time to build everything. Note that the initial build will download
+  a couple of files, like the official Raspberry Pi 3 firmware, the overlay root
+  fs etc. However, that is only done once, so subsequent builds won't re-download
+  them again (as long as you don't delete them).
+```
+$ make all -j4
+```
+
+- The last step is to partition and format the memory card and to put the files
+  onto the same. That is something we don't want to automate, since if anything
+  goes wrong, in worst case it might wipe one of your regular hard disks. Instead
+  what we have done, is that we have created another makefile target that will tell
+  you exactly what to do. Run that command and follow the instructions there.
+```
+$ make img-help
+```
+
+- Boot up the Pi. With all files on the memory card, put the memory card into
+the Raspberry Pi 3 and boot up the system. On the UART (we will add some wiring
+diagram soon, but until then, please use Google and search for UART on Raspberry
+Pi and you will get lots of guides) you will see the system booting up. When you
+have a shell, then it's simply just to follow the instructions on
+[here](https://github.com/OP-TEE/optee_os#6-load-driver-tee-supplicant-and-run-xtest)
+in the README.md to load tee-supplicant and run xtest.
+
+# 4. Known problems
+We encourage anyone interested in getting this into a better shape to help out.
+We have identified a couple issues while working with this. Some are harder to
+solve than others.
+
+## 4.1 Root file system
+Currently we are using a cpio archive with busybox as a base. However, we
+noticed that we didn't get any shell when booting up. This is probably easy to
+solve if one would spend some time looking into it. As a workaround, we have
+been putting an Ubuntu based root fs on top of the cpio archive now as an
+overlay. That works, but that also gives a lot of nasty warnings when booting
+up.
+
+Suggestions to look into regarding creating a better rootfs
+- Solve the issue with the no shell when using only the cpio archive.
+- Create a setup where one use [buildroot](https://buildroot.org) instead of
+  manually creating the cpio archive.
+- Create a 64bit [Raspbian](https://www.raspbian.org) image. This would be the
```

Comment:
I've read the same, but I guess it's just to wait and see. Suddenly there is someone with too much spare time that probably will do it.